### DISCUSSION TITLE
Creating dependencies between custom Op and with DBT Op

### DISCUSSION THREAD

**Discussion by IsaacChanXi** [dagster-io/dagster#19133] [Q&A] [answered]
Is there a way to create dependencies between DBT Op and a non-DBT Op?

It's possible to create such dependencies between DBT assets and non-DBT assets, but there could be scenarios where the non-DBT asset does not produce any data.

For example, before the DBT assets are triggered, I may like an Op that makes an API call to be triggered first. 

Any suggestions on what's the best approach to deal with such scenarios?

**Comment by OwenKephart** [comment, marked as answer]
Hi @IsaacChanXi -- generally if there's some code that you want to run before any dbt assets are executed, we recommend that you execute that code within the body of your `@dbt_assets` function, rather than putting it in a separate op, as it's not generally possible to make an asset depend on an op, and generally the utility of representing your dbt models as assets rather than an op is quite high.

Another option could be to represent your non-dbt op as an asset (even if it's not really one)

**Reply by IsaacChanXi** [reply]
Thanks, @OwenKephart, let me take a look into using @dbt_assets.

I presume your recommendation would be the same for non-DBT ops that depend on DBT assets?
